utils/src/main/java/com/cloud/utils/net/NetUtils.java

```diff
@@ -268,19 +268,7 @@ public static String getDefaultEthDevice() {
             final String defDev = Script.runSimpleBashScript("/sbin/route -n get default 2> /dev/null | grep interface | awk '{print $2}'");
             return defDev;
         }
-        final String defaultRoute = Script.runSimpleBashScript("/sbin/route | grep default");
-
-        if (defaultRoute == null) {
-            return null;
-        }
-
-        final String[] defaultRouteList = defaultRoute.split("\\s+");
-
-        if (defaultRouteList.length != 8) {
-            return null;
-        }
-
-        return defaultRouteList[7];
+        return Script.runSimpleBashScript("ip route | grep default | head -1 | grep -Po '(?<=dev )(\\S+)'");
```

Comment:
I would suggest:

<pre>
ip route show default 0.0.0.0/0|head -1|awk '{print $3}'
</pre>

If a Host has a large routing table you can run into problems here. By fetching the route for 0.0.0.0/0 you query the routing table faster.

Reply:
agrees. it should be better

Reply:
```ip route show default``` does not work with ip from brew on mac (which is not a big issue but would be nice.

Reply:
on mac
```
baloujak:var dahn$ ip route get default
default via 192.168.144.99 dev en0
baloujak:var dahn$ ip route get default | awk '{print $5}'
en0
```
have to test if that would work on other platforms

Reply:
kvm-centos7-kvm1 ~]# ip route get default
need at least a destination address

Reply:
I don't know where that commen t went but @wido we need the device on which the default route is so we need to {print $5} not $3

Reply:
While there might be some speed improvements in Wido's comment, I don't expect one is going to have tons of routes on a blank management server (where the needed code we are patching is being executed), and I definitively will not have time testing it all over again.

Reply:
I don't mind that much. I would just think about these things are you don't need to print the whole routing table.

Especially on a Hypervisor/Host this is something to think of.

Reply:
@wido , I do recall that ip route work as part of refactoring VXLAN scripts, so that makes sense perfectly.

Reply:
If speed is a requirement, I won't bother iproute2, just use `procfs`: `awk '$2 == 00000000 { print $1 }' /proc/net/route`

Reply:
I'd just stick with ip route. That's good and reliable.

You might want to add the '-o' flag which makes it better to parse.